modules/market/src/main/java/com/opengamma/strata/market/surface/DefaultSurfaceMetadataBuilder.java

```diff
@@ -0,0 +1,277 @@
+   */
+  private ValueType xValueType = ValueType.UNKNOWN;
+  /**
+   * The y-value type, providing meaning to the y-values of the surface.
+   * <p>
+   * This type provides meaning to the y-values.
+   * <p>
+   * If using the builder, this defaults to {@link ValueType#UNKNOWN}.
+   */
+  private ValueType yValueType = ValueType.UNKNOWN;
+  /**
+   * The y-value type, providing meaning to the z-values of the surface.
+   * <p>
+   * This type provides meaning to the z-values.
+   * <p>
+   * If using the builder, this defaults to {@link ValueType#UNKNOWN}.
+   */
+  private ValueType zValueType = ValueType.UNKNOWN;
+  /**
+   * The additional surface information.
+   * <p>
+   * This stores additional information for the surface.
+   */
+  private final Map<SurfaceInfoType<?>, Object> info = new HashMap<>();
+  /**
+   * The metadata about the parameters.
+   * <p>
+   * If present, the parameter metadata will match the number of parameters on the surface.
+   */
+  private List<SurfaceParameterMetadata> parameterMetadata;
+
+  /**
+   * Restricted constructor.
+   */
+  DefaultSurfaceMetadataBuilder() {
+  }
+
+  /**
+   * Restricted copy constructor.
+   * 
+   * @param beanToCopy  the bean to copy from
+   */
+  DefaultSurfaceMetadataBuilder(DefaultSurfaceMetadata beanToCopy) {
+    this.surfaceName = beanToCopy.getSurfaceName();
+    this.xValueType = beanToCopy.getXValueType();
+    this.yValueType = beanToCopy.getYValueType();
+    this.zValueType = beanToCopy.getZValueType();
+    this.info.putAll(beanToCopy.getInfo());
+    this.parameterMetadata = beanToCopy.getParameterMetadata().map(m -> new ArrayList<>(m)).orElse(null);
+  }
+
+  //-----------------------------------------------------------------------
+  /**
+   * Sets the surface name.
+   * 
+   * @param surfaceName  the surface name
+   * @return this, for chaining
+   */
+  public DefaultSurfaceMetadataBuilder surfaceName(String surfaceName) {
+    ArgChecker.notNull(surfaceName, "surfaceName");
+    this.surfaceName = SurfaceName.of(surfaceName);
+    return this;
+  }
+
+  /**
+   * Sets the surface name.
+   * 
+   * @param surfaceName  the surface name
+   * @return this, for chaining
+   */
+  public DefaultSurfaceMetadataBuilder surfaceName(SurfaceName surfaceName) {
+    ArgChecker.notNull(surfaceName, "surfaceName");
+    this.surfaceName = surfaceName;
+    return this;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Sets the x-value type, providing meaning to the x-values of the surface.
+   * <p>
+   * This type provides meaning to the x-values.
+   * <p>
+   * If using the builder, this defaults to {@link ValueType#UNKNOWN}.
+   * 
+   * @param xValueType  the x-value type
+   * @return this, for chaining
+   */
+  public DefaultSurfaceMetadataBuilder xValueType(ValueType xValueType) {
+    ArgChecker.notNull(xValueType, "xValueType");
+    this.xValueType = xValueType;
```

Comment:
These 2 lines could be combined as

    this.xValueType = ArgChecker.notNull(xValueType, "xValueType");

That's the pattern we've used elsewhere.